test/Container/ApplicationFactoryTest.php

```diff
@@ -1071,4 +1071,226 @@ public function testCanSpecifyRouteOptionsViaConfiguration()
         $this->assertInstanceOf(Route::class, $route);
         $this->assertEquals($expected, $route->getOptions());
     }
+
+    /**
+     * @expectedException \Zend\Expressive\Container\Exception\InvalidArgumentException
+     */
+    public function testExceptionIsRaisedInCaseOfInvalidRouteMethodsConfiguration()
+    {
+        $router = $this->prophesize('Zend\Expressive\Router\RouterInterface');
```

Comment:
Oops. After #195 is merged this should be refactored.

You can rebase your changes against my fork.